Assets/Scripts/UI/InGame/Graphics/GeneralGraphics.cs

```diff
@@ -0,0 +1,287 @@
+		[SerializeField] public Dropdown shadows;
+		[SerializeField] public Toggle vSync;
+		[SerializeField] public Toggle softParticles;
+
+		[SerializeField] public Toggle customSettings;
+
+		public Dropdown TextureQuality
+		{ 
+			get { return textureQuality; }
+			set { textureQuality = value; }
+		}
+		public Dropdown ShadowRes
+		{ 
+			get { return shadowRes; }
+			set { shadowRes = value; }
+		}
+		public Dropdown AntiAliasing
+		{
+			get { return antiAliasing; }
+			set { antiAliasing = value; }
+		}
+		public Dropdown Shadows
+		{
+			get { return shadows; }
+			set { shadows = value; }
+		}
+		public Toggle VSync
+		{
+			get { return vSync; }
+			set { vSync = value; }
+		}
+		public Toggle SoftParticles
+		{
+			get { return softParticles; }
+			set { softParticles = value; }
+		}
+
+		public Toggle CustomSettings
+		{
+			get { return customSettings; }
+			set { customSettings = value; }
+		}
+
+		private void Update() {
+			
+			if(!CustomSettings.isOn)
+			{
+				// shadow res
+				if(QualitySettings.shadowResolution == ShadowResolution.Low)
+				{
+					ShadowRes.value = 0;
+				}
+				else if(QualitySettings.shadowResolution == ShadowResolution.Medium)
+				{
+					ShadowRes.value = 1;
+				}
+				else if(QualitySettings.shadowResolution == ShadowResolution.High)
+				{
+					ShadowRes.value = 2;
+				}
+				else if(QualitySettings.shadowResolution == ShadowResolution.VeryHigh)
+				{
+					ShadowRes.value = 3;
+				}
+
+				// shadows
+				if(QualitySettings.shadows == ShadowQuality.Disable)
+				{
+					Shadows.value = 0;
+				}
+				else if(QualitySettings.shadows == ShadowQuality.HardOnly)
+				{
+					Shadows.value = 1;
+				}
+				else if(QualitySettings.shadows == ShadowQuality.All)
+				{
+					Shadows.value = 2;
+				}
+
+				// texture quality
+				if(QualitySettings.masterTextureLimit == 3)
+				{
+					TextureQuality.value = 0;
+				}
+				else if(QualitySettings.masterTextureLimit == 2)
+				{
+					TextureQuality.value = 1;
+				}
+				else if(QualitySettings.masterTextureLimit == 1)
+				{
+					TextureQuality.value = 2;
+				}
+				else if(QualitySettings.masterTextureLimit == 0)
+				{
+					TextureQuality.value = 3;
+				}
+				
+				// anti aliasing
+				if(QualitySettings.antiAliasing == 0)
+				{
+					AntiAliasing.value = 0;
+				}
+				else if(QualitySettings.antiAliasing == 2)
+				{
+					AntiAliasing.value = 1;
+				}
+				else if(QualitySettings.antiAliasing == 4)
+				{
+					AntiAliasing.value = 2;
+				}
+				else if(QualitySettings.antiAliasing == 8)
+				{
+					AntiAliasing.value = 3;
+				}
+
+				// soft particles
+				if(QualitySettings.softParticles)
+				{
+					SoftParticles.isOn = true;
+				}
+				else
+				{
+					SoftParticles.isOn = false;
+				}
+
+				// vsync
+				if(QualitySettings.vSyncCount == 0)
+				{
+					VSync.isOn = false;
+				}
+				else
+				{
+					VSync.isOn = true;
+				}
+			}
+		}
+		public void ChangeShadows()
+		{
+			var selected = Shadows.value;
```

Comment:
We can convert enums to integers, but we can also convert integers to enums.
`QualitySettings.shadows = (ShadowQuality) Shadows.value`